src/mfs.jl

```diff
@@ -0,0 +1,230 @@
+
+    for _ in 1:total_iterations
+        generator, _ = statespace_sampler(GLOBAL_RNG;
+                                         radius = 1.0, spheredims = dim, 
+                                         center = zeros(dim) ) 
+        perturbation = generator() * temp_dist
+        new_shock = perturbation + X
+        
+        if !(id_x == mapper(new_shock))
+            best_dist = norm(perturbation)
+            best_shock = perturbation
+            temp_dist = best_dist-best_dist/delta
+        end
+    end
+
+    return best_shock, best_dist
+end
+
+
+
+
+function mfs_objective(perturbation, X, idX, mapper::AttractorMapper, penalty=1000.0) 
+
+    dist = norm(perturbation)
+    if dist == 0
+        return penalty
+    end
+    if mapper(X + perturbation) == idX
+        return dist + penalty
+    else
+        return dist
+    end
+end
+
+
+"""
+    minimal_fatal_shock(mapper::AttractorMapper, X, search_area, algorithm::Union{MFSBruteForce, MFSBlackBoxOptim})  
+
+Runs the minimal fatal shock algorithm on the initial point X. Two algorithms are available: MFSBruteForce and MFSBlackBoxOptim.
+MFSBruteForce is an algorithm based on randomised search with two consequent steps: random initialization and sphere radius reduction.
+MFSBlackBoxOptim is an algorithm based derivative free optimization. It uses BlackBoxOptim package to find the best shock.
+
+`mapper::AttractorMapper` one of available in Attractors.jl [`AttractorMapper`](@ref) constructed with respect to dynamical system.
+`X` vector containing coordinates of initial point to be tested.
+`search_area` array of tuples, by specifying single value e.g. [(-1.5, 1.5)] you specify this value for each of the variables, or you may specify separetely 
+search area for each of your states by, for example in 2 dimensions, [(-1.5, 1.5), (-1, 1)].
+`algorithm` one of the two algorithms available: `MFSBruteForce` or `MFSBlackBoxOptim`.
+
+## `MFSBruteForce`
+`initial_iterations` number of random pertubations to try in the first step of the algorithm, default = 10000.
+`sphere_iterations` number of steps while initializing random points on hypersphere and decreasing its radius, default = 10000.
+`dimension` dimension of the system, default = 2.
+`sphere_decrease_step` factor by which the radius of the hypersphere is decreased, default = 1000.0, the higher the value, 
+the more precise values may be obtained.
+* Default parameters initialization `algorithm = MFSBruteForce()` will set parameters to:
+    - `initial_iterations = 10000`
+    - `sphere_iterations = 10000`
+    - `dimension = 2`
+    - `sphere_decrease_step = 100.0`
+
+## `MFSBlackBoxOptim`
+`dimension` dimension of the system.
+`guess` vector of initial guesses for the optimization algorithm.
+`penalty` penalty value for the objective function.
+* Default parameters initialization `algorithm = MFSBlackBoxOptim()` will set default parameters:
+    - `dimension = 2`
+    - `guess = []`
+    - `penalty = 1000.0`
```

Comment:
Why do you repeat the keywords now but with their defaults? Doesn't it make more concise to do both at the same time as 

- `penalty = 100.0`: penalty value for object function (explain what this actually means)